src/sentry/uptime/models.py

```diff
@@ -281,3 +290,34 @@ def load_regions_for_uptime_subscription(
 
 class UptimeRegionScheduleMode(enum.StrEnum):
     ROUND_ROBIN = REGIONSCHEDULEMODE_ROUND_ROBIN
+
+
+@data_source_type_registry.register(DATA_SOURCE_UPTIME_SUBSCRIPTION)
+class UptimeSubscriptionDataSourceHandler(DataSourceTypeHandler[UptimeSubscription]):
+    @staticmethod
+    def bulk_get_query_object(
+        data_sources: list[DataSource],
+    ) -> dict[int, UptimeSubscription | None]:
+        uptime_subscription_ids: list[UUID] = []
+
+        for ds in data_sources:
+            try:
+                subscription_id = UUID(ds.source_id)
+                uptime_subscription_ids.append(subscription_id)
+            except ValueError:
+                logger.exception(
+                    "Invalid DataSource.source_id fetching UptimeSubscription",
+                    extra={"id": ds.id, "source_id": ds.source_id},
+                )
+
+        qs_lookup = {
+            str(uptime_subscription.subscription_id): uptime_subscription
+            for uptime_subscription in UptimeSubscription.objects.filter(
+                subscription_id__in=uptime_subscription_ids
+            )
+        }
```

Comment:
Might be a good idea to log an error if we don't return a subscription for every id?

Reply:
will follow up 👍